hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/recon/TestRecon.java

```diff
@@ -0,0 +1,395 @@
+        + "/api/utilization";
+    taskStatusURL = "http://" + reconHTTPAddress
+        + "/api/task/status";
+
+    cluster =  MiniOzoneCluster.newBuilder(conf).build();
+    cluster.waitForClusterToBeReady();
+    metadataManager = cluster.getOzoneManager().getMetadataManager();
+
+    cluster.getStorageContainerManager().exitSafeMode();
+    reconUtils = new ReconUtils();
+
+    // initialize HTTPClient
+    httpClient = HttpClientBuilder
+        .create()
+        .setDefaultRequestConfig(config)
+        .build();
+
+    pauseInterval = conf.getTimeDuration(
+        RECON_OM_SNAPSHOT_TASK_INTERVAL,
+        RECON_OM_SNAPSHOT_TASK_INTERVAL_DEFAULT,
+        TimeUnit.MILLISECONDS);
+  }
+
+  @AfterClass
+  public static void shutdown() {
+    if (cluster != null) {
+      cluster.shutdown();
+    }
+  }
+
+  /**
+   * Returns a {@link CloseableHttpClient} configured by given configuration.
+   * If conf is null, returns a default instance.
+   *
+   * @param url        URL
+   * @return a JSON String Response.
+   */
+  private String makeHttpCall(String url)
+      throws IOException {
+    HttpGet httpGet = new HttpGet(url);
+    HttpResponse response = httpClient.execute(httpGet);
+    int errorCode = response.getStatusLine().getStatusCode();
+    HttpEntity entity = response.getEntity();
+
+    if ((errorCode == HTTP_OK) || (errorCode == HTTP_CREATED)) {
+      return EntityUtils.toString(entity);
+    }
+
+    if (entity != null) {
+      throw new IOException("Unexpected exception when trying to reach Ozone " +
+          "Manager, " + EntityUtils.toString(entity));
+    } else {
+      throw new IOException("Unexpected null in http payload," +
+          " while processing request");
+    }
+  }
+
+  @Test
+  public void testReconServer() throws Exception {
+    //add a vol, bucket and key
+    addKeys(0, 1);
+
+    //check if OM metadata has vol0/bucket0/key0 info
+    String ozoneKey = metadataManager.getOzoneKey(
+        "vol0", "bucket0", "key0");
+    OmKeyInfo keyInfo1 = metadataManager.getKeyTable().get(ozoneKey);
+
+    TableIterator<String, ? extends Table.KeyValue<String, OmKeyInfo>>
+        omKeyValueTableIterator = metadataManager.getKeyTable().iterator();
+
+    long omMetadataKeyCount = getTableKeyCount(omKeyValueTableIterator);
+
+    //verify if OM has /vol0/bucket0/key0
+    Assert.assertEquals("vol0", keyInfo1.getVolumeName());
+    Assert.assertEquals("bucket0", keyInfo1.getBucketName());
+
+    //pause to get the next snapshot from om
+    Thread.sleep(pauseInterval);
+
+    // HTTP call to /api/containers
+    String containerResponse = makeHttpCall(containerKeyServiceURL);
+    Map map = new Gson().fromJson(containerResponse, HashMap.class);
+    LinkedTreeMap linkedTreeMap = (LinkedTreeMap) map.get("data");
+    long reconMetadataKeyCount = (long)(double)linkedTreeMap.get("totalCount");
+
+    //verify count of keys after full snapshot
+    Assert.assertEquals(omMetadataKeyCount, reconMetadataKeyCount);
```

Comment:
Here, we are actually comparing number of keys in OM and number of containers in Recon containers API endpoint. In this case it happens to be true because we associate each key with its own container. Is there a way we can compare keys to keys?